src/transformers/models/realm/modeling_realm.py

```diff
@@ -1543,6 +1542,7 @@ def forward(
         head_mask=None,
         inputs_embeds=None,
         relevance_score=None,
+        block_mask=None,
```

Comment:
It seems like `token_type_ids` is not used here anymore at all. This is quite backwards compatibility breaking. Is it intended? Should `block_mask` replace `token_type_ids`?

Reply:
`token_type_ids` is still used in the realm(bert) model inside `RealmReader`. We just don't use `token_type_ids` as the candidate mask for convenience and use the newly added `block_mask` instead, because I realized the former approach would result in `[SEP]` token occurring in the predicted answer spans, which is explained [here](https://github.com/huggingface/transformers/pull/15297#issue-1111824123).

This is indeed a small breaking change. However, since users would not directly leverage `RealmReader` (`RealmReader` 
is an inner model of `RealmForOpenQA`, and the `relevance_score` input of `RealmReader` needs to be firstly computed in `RealmForOpenQA`), IMO I think this would not affect existing codes.